scripts/o3de/tests/test_repo_properties.py

```diff
@@ -0,0 +1,180 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+import json
+import pytest
+import pathlib
+from unittest.mock import patch
+
+from o3de import repo_properties
+
+TEST_TEMPLATE_REPO_JSON = '''
+{
+    "repo_name": "repotest",
+    "repo_uri": "https://test",
+    "$schemaVersion": "1.0.0",
+    "origin": "o3de",
+    "origin_url": "${OriginURL}",
+    "summary": "${Summary}",
+    "additional_info": "${AdditionalInfo}",
+    "last_updated": "",
+    "gems_data": [],
+    "projects_data": [],
+    "templates_data": []
+}
+'''
+# class returns a dictionary of the template repo json
+@pytest.fixture(scope='function')
+def init_repo_json_data(request):
+    class RepoJsonData:
+        def __init__(self):
+            self.data = json.loads(TEST_TEMPLATE_REPO_JSON)
+            self.path = None
+    request.cls.repo_json = RepoJsonData()
+
+# use the TEST_TEMPLATE_REPO_JSON created above as the repo json template
+@pytest.mark.usefixtures('init_repo_json_data')
+class TestEditRepoProperties:
+
+    @staticmethod
+    def resolve(self):
+        return self
+
+    @pytest.mark.parametrize("repo_path, repo_name, \
```

Comment:
The repo_path and repo_name are the same in all the tests, so I recommend removing them from the params or modify some of the tests to use different paths and names